Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -0,0 +1,157 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.WrapPanel
+{
+    /// <summary>
+    /// WrapPanel is a panel that position child control vertically or horizontally based on the orientation and when max width/ max height is recieved a new row(in case of horizontal) or column (in case of vertical) is created to fit new controls.
+    /// </summary>
+    public partial class WrapPanel : Panel
+    {
+        private List<double> _vSize;
+
+        /// <summary>
+        /// Gets or sets the orientation of the WrapPanel, Horizontal or vertical means that child controls will be added horizontally until the width of the panel can't fit more control then a new row is added to fit new horizontal added child controls, vertical means that child will be added vertically until the height of the panel is recieved then a new column is added
+        /// </summary>
+        public Orientation Orientation
+        {
+            get { return (Orientation)GetValue(OrientationProperty); }
+            set { SetValue(OrientationProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Orientation"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty OrientationProperty =
+            DependencyProperty.Register(
+                "Orientation",
+                typeof(Orientation),
+                typeof(WrapPanel),
+                new PropertyMetadata(Orientation.Horizontal));
+
+        /// <inheritdoc />
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            _vSize = new List<double> { 0.0 };
+            var totalMeasure = new UvMeasure();
+            var parentMeasure = new UvMeasure(Orientation, availableSize.Width, availableSize.Height);
+            var lineMeasure = new UvMeasure();
+            foreach (var child in Children)
+            {
+                child.Measure(availableSize);
+
+                var currentMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
+
+                if (parentMeasure.U > currentMeasure.U + lineMeasure.U)
```

Comment:
Will assume that the orientation is horizontal

// this line is invalid because I'm not considered with the parent measure while placing the controls I'm concerned with the max width so the panel can draw the child correctly
totalMeasure.U = Math.Max(lineMeasure.U, parentMeasure.U);

inside the else we need 2 conditions your code ignores the first check what if the row can handle more controls ?
in your code you already added the currentMeasure to the line measure before the if condition which is not required to be add in case we need to add a new row and the current control can take more controls with it :) 

in the first else statement I add measure to the _vSize twice
1) a new line should be added so added the line measure.
2) inside the second else statement when we added a new row and the current control will take the whole row I add it's height.
so the height is added twice but in your code it is added only once and you can't count on the  _vSize.Add(lineMeasure.V); outside the for loop because this will add it in case the control is the last on the loop but what would happen in case the control is the first or second ?

I do enjoy having you as a reviewer :)